Source/WebCore/platform/graphics/WidthIterator.cpp

```diff
@@ -230,17 +222,35 @@ static void addToGlyphBuffer(GlyphBuffer& glyphBuffer, Glyph glyph, const Font&
         glyphBuffer.add(0, font, 0, currentCharacterIndex + 1);
 }
 
+static const Font* fontForRange(const Font* font, const Font* synthesizedFont, const Font* smallSynthesizedFont, bool isSmallCaps)
+{
+    if (synthesizedFont)
+        return isSmallCaps ? smallSynthesizedFont : synthesizedFont;
+    return font;
+}
+
 template <typename TextIterator>
 inline void WidthIterator::advanceInternal(TextIterator& textIterator, GlyphBuffer& glyphBuffer)
 {
     // The core logic here needs to match FontCascade::widthForSimpleText()
     bool rtl = m_run.rtl();
 
+    GlyphBuffer rangeBuffer;
```

Comment:
Why did you opt to use a staging GlyphBuffer, and then copy its contents into the real glyph buffer during the commit step, rather than just using the original GlyphBuffer directly?

Reply:
🆒